apis/Google.Cloud.Spanner.Data/docs/configuration.md

```diff
@@ -0,0 +1,180 @@
+large numbers of sessions all being refreshed at exactly the same
+time.
+
+Applications rarely need to change this setting.
+
+**PoolEvictionDelay**
+
+Default value: 7 days
+
+Processes that run for extended periods may refresh sessions many
+times, and the server will generally permit this indefinitely.
+However, that puts a strain on server-side resource management. This
+setting controls the time after which sessions are evicted. Like
+refresh settings, jitter is applied so that sessions are not evicted
+all at once.
+
+Applications rarely need to change this setting.
+
+**WaitOnResourcesExhausted**
+
+Default value: Block
+
+This enum value controls the behavior when an application requests a
+new session despite having already acquired the maximum number of
+active sessions.
+
+The default value (`Block`) will block the pending
+session acquisition operation until a session is available. This
+allows an application to continue to function correctly, but with
+reduced responsiveness.
+
+The value of `Fail` will cause the pending session acquisition to
+throw an exception. This is useful for spotting resource leaks, and
+some applications may prefer a swift failure over a delayed response
+when the application is heavily loaded.
+
+**Timeout**
+
+Default value: 60
+
+The total time in seconds allowed for a network call to the Cloud
+Spanner server, including retries. This setting is applied to calls
+to create, refresh and delete sessions, as well as beginning
+transactions.
+
+Applications rarely need to change this setting
+
+**MaximumConcurrentSessionCreates**
+
+Default value: 10
+
+Spanner has limits on the number of sessions that can be created
+concurrently without affecting performance. The session pool uses
+this setting to throttle the number of concurrent session creation
+operations, across all databases.
+
+Applications rarely need to change this setting.
+
+## Modifying the default SessionPoolOptions from ADO.NET
+
+When using the ADO.NET implementation (Google.Cloud.Spanner.Data), a
+`SessionPoolManager` is used to create `SessionPool` instances based
+on the connection string. (Different endpoints would require
+different pools, for example.) Each `SessionPoolManager` has a set
+of `SessionPoolOptions` that it uses whenever it creates a new
+`SessionPool`. It's easy to modify the options used by
+the default `SessionPoolManager`. For example, to change the minimum
+number of sessions in each session pool:
+
+```csharp
+SessionPoolManager.Default.SessionOptions.MinimumPooledSessions = 100;
+```
+
+Any changes to options should be made before the Spanner operation.
+Changes to options after a `SessionPool` has become active *may*
+still take effect, but that varies by option and may change over time.
+
+## Using a non-default SessionPoolManager
+
+An alternative to modifying the options used by the default
+`SessionPoolManager` to create a new manager with a new set of
```

Comment:
nit: _is_ to create

Reply:
Done, and also added "object" to make it clearer I'm not suggesting that people write their own classes.